docs/proposals/workflow.md

```diff
@@ -0,0 +1,134 @@
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Abstract
+
+A proposal to modify [`Job` resource](../../docs/user-guide/jobs.md) to implement a minimal [Workflow managment system](https://en.wikipedia.org/wiki/Workflow_management_system) in kubernetes.
+Workflows (aka DAG workflows since jobs are organized in a Direct Acyclic Graph) are ubiquitous in modern [job schedulers](https://en.wikipedia.org/wiki/Job_scheduler), see for example:
+    - [luigi](https://github.com/spotify/luigi)
+    - [ozie](http://oozie.apache.org/)
+    - [azkaban](https://azkaban.github.io/)
+
+Most of the [job schedulers](https://en.wikipedia.org/wiki/List_of_job_scheduler_software) offer workflow functionality to some extent.
+
+## Use Cases
+
+    * As a user I want to be able to define job chains such that the completion of a given job will trigger other jobs.
+
+## Implementation
+
+The basic idea consists in adding a label selector to the current Job API object. The new selector will determine the parent jobs.  The parent jobs are the jobs current job will depend on. The current job will be scheduled once all the parent jobs will run to completion.
+The strongest point in favor of this approach is the ability to re-use the current `job` implementation. Using a label selector the current `job` will simply become a `graph` of jobs: all the features/controllers that will be built on top of `job` like for example [#11980](https://github.com/kubernetes/kubernetes/issues/11980) and [#17242](https://github.com/kubernetes/kubernetes/issues/17242) will be automatically extended to support workflows. Not sure about https://github.com/kubernetes/kubernetes/issues/16845 and https://github.com/kubernetes/kubernetes/issues/14188.
+A similar approach is implemented in Chronos for [dependent jobs](https://mesos.github.io/chronos/docs/api.html#adding-a-dependent-job).
+
+### API Object
+
+To implement _workflows_ the `Job` API should be modified:
+
+```go
+// JobsSelector it's just an alias for PodSelector
+type JobSelector PodSelector
+
+// Job represents the configuration of a single job.
+type Job struct {
+    unversioned.TypeMeta `json:",inline"`
+    // Standard object's metadata.
+
+    // More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+    api.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec is a structure defining the expected behavior of a job.
+    // More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+    Spec JobSpec `json:"spec,omitempty"`
+
+    // Status is a structure describing current status of a job.
+    // More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+    Status JobStatus `json:"status,omitempty"`
+
+    // JobSelector to detect parents jobs
+    ParentSelector *JobSelector `json:"parentSelector, omitempty"`
```

Comment:
I guess you've meant here PodSelector and not JobSelector, btw. looking at PodSelector I think we should generalize this so it's applicable as well here. I'll open a separate issue for that.